krypto/src/commonMain/kotlin/com/soywiz/krypto/encoding/Base64.kt

```diff
@@ -1,42 +1,58 @@
 package com.soywiz.krypto.encoding
 
 object Base64 {
-    private val TABLE = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/="
-    private val DECODE = IntArray(0x100).apply {
+    private const val TABLE = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/="
+    private const val TABLE_SAFE = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_="
+
+    private val decoder = IntArray(0x100).apply {
         for (n in 0..255) this[n] = -1
-        for (n in 0 until TABLE.length) {
+        for (n in TABLE.indices) {
             this[TABLE[n].toInt()] = n
         }
     }
 
+    private val decoderUrlSafe = IntArray(0x100).apply {
```

Comment:
Though I saw that initially this was not done, maybe IntArray(0x100) { -1 } would remove the requirement for the first for on both cases. Not a big body block, but maybe this table could be generated in a private function receiving the table so you can deduplicate that code

Reply:
Done